JVM GC ReferenceQueue should not be flagged as blocking

### Describe the bug

When creating an object that is placed into a ReferenceQueue by the JVM it acquires a ReentrantLock. We should ignore this for blocking purposes.

Example stack trace:

```
java.lang.AssertionError: Blocking call! jdk.internal.misc.Unsafe#park on thread Thread[#102,non-blocking-thread-WriteSkewCacheLoaderFunctionalTest-NodeA-p14-t3,5,ISPN-non-blocking-group]
	at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIntegration.java:51)
	at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
	at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
	at java.base/jdk.internal.misc.Unsafe.park(Unsafe.java)
	at java.base/java.util.concurrent.locks.LockSupport.park(LockSupport.java:221)
	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:754)
	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:990)
	at java.base/java.util.concurrent.locks.ReentrantLock$Sync.lock(ReentrantLock.java:153)
	at java.base/java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:322)
	at java.base/java.lang.ref.ReferenceQueue.poll(ReferenceQueue.java:182)
	at java.base/jdk.internal.vm.ThreadContainers.expungeStaleEntries(ThreadContainers.java:72)
	at java.base/jdk.internal.vm.ThreadContainers.registerContainer(ThreadContainers.java:89)
	at java.base/jdk.internal.vm.SharedThreadContainer.create(SharedThreadContainer.java:85)
	at java.base/jdk.internal.vm.SharedThreadContainer.create(SharedThreadContainer.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.&lt;init&gt;(ThreadPoolExecutor.java:1322)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.&lt;init&gt;(ScheduledThreadPoolExecutor.java:512)
	at org.infinispan.commons.executors.ScheduledThreadPoolExecutorFactory.createExecutor(ScheduledThreadPoolExecutorFactory.java:19)
	at org.infinispan.commons.executors.ScheduledThreadPoolExecutorFactory.createExecutor(ScheduledThreadPoolExecutorFactory.java:12)
	at org.infinispan.executors.LazyInitializingScheduledExecutorService.initIfNeeded(LazyInitializingScheduledExecutorService.java:49)
	at org.infinispan.executors.LazyInitializingScheduledExecutorService.schedule(LazyInitializingScheduledExecutorService.java:150)
	at org.infinispan.topology.LocalTopologyManagerImpl.withView(LocalTopologyManagerImpl.java:749)
	at org.infinispan.topology.LocalTopologyManagerImpl.handleStatusRequest(LocalTopologyManagerImpl.java:367)
	at org.infinispan.commands.topology.CacheStatusRequestCommand.invokeAsync(CacheStatusRequestCommand.java:43)
	at org.infinispan.topology.TopologyManagementHelper.invokeAsync(TopologyManagementHelper.java:152)
	at org.infinispan.topology.TopologyManagementHelper.executeOnClusterSync(TopologyManagementHelper.java:52)
	at org.infinispan.topology.ClusterTopologyManagerImpl.fetchClusterStatus(ClusterTopologyManagerImpl.java:590)
	at org.infinispan.topology.ClusterTopologyManagerImpl.recoverClusterStatus(ClusterTopologyManagerImpl.java:466)
	at org.infinispan.topology.ClusterTopologyManagerImpl.lambda$handleClusterView$6(ClusterTopologyManagerImpl.java:431)
	at org.infinispan.util.concurrent.ActionSequencer.safeNonBlockingCall(ActionSequencer.java:59)
	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.run(ActionSequencer.java:228)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)

```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/commons/all/src/main/java/org/infinispan/commons/internal/CommonsBlockHoundIntegration.java
+++ b/commons/all/src/main/java/org/infinispan/commons/internal/CommonsBlockHoundIntegration.java
@@ -1,5 +1,6 @@
 package org.infinispan.commons.internal;
 
+import java.lang.ref.ReferenceQueue;
 import java.lang.reflect.Method;
 import java.security.SecureRandom;
 import java.util.concurrent.ArrayBlockingQueue;
@@ -43,6 +44,10 @@ public class CommonsBlockHoundIntegration implements BlockHoundIntegration {
 
       // BoundedLocalCache is unfortunately package private
       builder.allowBlockingCallsInside("com.github.benmanes.caffeine.cache.BoundedLocalCache", "performCleanUp");
+
+      // ReferenceQueue in JVM can cause blocking in some cases - we are assuming users of this are responsible and the
+      // lock should be short lived even if contested
+      builder.allowBlockingCallsInside(ReferenceQueue.class.getName(), "poll");
 
       handleJREClasses(builder);
 

